metrics/metrics.go

```diff
@@ -0,0 +1,93 @@
+package metrics
+
+import (
+	"context"
+	"net"
+	"net/http"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/syncint64"
+	"go.opentelemetry.io/otel/sdk/metric"
+)
+
+var (
+	log = logging.Logger("metrics")
+)
+
+type Metrics struct {
+	Exporter *prometheus.Exporter
+
+	GetValueKeysLatency      syncint64.Histogram
+	GetMetadataLatency       syncint64.Histogram
+	PutMultihashBatchLatency syncint64.Histogram
+	PutMetadataLatency       syncint64.Histogram
+
+	s *http.Server
+}
+
+func New(metricsAddr string) (*Metrics, error) {
+
+	m := &Metrics{}
+
+	var err error
+	if m.Exporter, err = prometheus.New(prometheus.WithoutUnits()); err != nil {
+		return nil, err
+	}
+
+	provider := metric.NewMeterProvider(metric.WithReader(m.Exporter))
+	meter := provider.Meter("ipni/dhstore")
+
+	if m.GetValueKeysLatency, err = meter.SyncInt64().Histogram("ipni/dhstore/get_value_keys_latency",
+		instrument.WithUnit("milliseconds"),
+		instrument.WithDescription("Find ValueKeys by a Multihash latency")); err != nil {
+		return nil, err
+	}
+	if m.GetMetadataLatency, err = meter.SyncInt64().Histogram("ipni/dhstore/get_metadata_latency",
+		instrument.WithUnit("milliseconds"),
+		instrument.WithDescription("Find Metadata by a ValueKey latency")); err != nil {
+		return nil, err
+	}
+
+	if m.PutMultihashBatchLatency, err = meter.SyncInt64().Histogram("ipni/dhstore/put_multihash_batch_latency",
+		instrument.WithUnit("milliseconds"),
+		instrument.WithDescription("Put ValueKey for a batch of Multihashes latency")); err != nil {
+		return nil, err
+	}
+	if m.PutMetadataLatency, err = meter.SyncInt64().Histogram("ipni/dhstore/put_metadata_latency",
+		instrument.WithUnit("microseconds"),
+		instrument.WithDescription("Put Metadata for a ValueKey latency")); err != nil {
+		return nil, err
+	}
```

Comment:
I think we can reduce the verbosity of metrics by using a single latency metric, say `ipni/dhstore/http_latency` that is tagged by: `method:<http-method>`, `path:<http-path-prefix>, and `status:<http-status-code>`. 

That would make up a more concise and graphing a bit easier on grafana.